### PR TITLE
🐛 test/e2e: don't restart the conformance test container after ginkgo exits

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Currently the e2e tests which are running the conformance tests are failing.

E.g. https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-cluster-api-e2e-workload-upgrade-1-18-1-19-release-1-3/1615714183762939904

We introduced a new validation in https://github.com/kubernetes-sigs/cluster-api/pull/7928 which ensures that all containers have been stopped after the e2e test finishes.

This validation discovered that when we run conformance tests the conformance container is still "running" even after the conformance tests are done. This is because after the conformance tests / ginkgo exits Docker will just restart the conformance test container and the conformance tests will run again.

The reason for this is that we are using "unless-stopped" for all the containers we start. This PR changes the behavior to never restart the conformance container after it exits.

Surfaced via this Slack thread: https://kubernetes.slack.com/archives/C8TSNPY4T/p1674057996218879

Thx @aniruddha2000 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
/test pull-cluster-api-e2e-full-main
/test pull-cluster-api-e2e-workload-upgrade-1-26-latest-main

/hold
I want to wait for the upgrade test to succeed before we merge (this validates that the fix works)

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
Teste locally, works on my machine

**Comment by fabriziopandini** [comment]
Nice catch @aniruddha2000 and @sbueringer!
/lgtm
/approve

**Review by ykakarap — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
/hold cancel

Test green

**Pull request merged**
